config/components/single-namespace/kustomization.yaml

```diff
@@ -32,8 +32,8 @@ replacements:
         kind: MutatingWebhookConfiguration
       fieldPaths:
       - webhooks.0.clientConfig.service.namespace
-patchesStrategicMerge:
-- |-
+patches:
+- patch: |-
```

Comment:
The patch contains the `$patch: delete` directive, which is not valid in Json6902. This produce the following error when running `make single-up`:
```
Error: accumulating components: accumulateDirectory: "couldn't make target for path
'k8ssandra-operator/config/components/single-namespace': invalid Kustomization: json: unknown field
\"$patch\""
error: no objects passed to apply
make: *** [single-up] Error 1
```
I'm looking for the equivalent syntax...

Reply:
I can't find a way to delete the namespace in a JSON 6902 patch. There is `op: remove` but it requires a path to delete a particular element within the resource.

Alternatively, we could label the namespace, and then filter out the label in the `kubectl apply` command with the `-l` option. Or simply run an additional `kubectl delete` command after the kustomization in the Makefile.

Reply:
Kustomization should detect whether it's SM or JSON6902 patch automatically. It seems the detection is failing here then..